### PR TITLE
Integration Tests for mac_xattr.py

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?

Add's integration tests for `mac_xattr.py` execution module
### What issues does this PR fix or reference?

https://github.com/saltstack/zh/issues/553
### Tests written?

Yes

### CODE REVIEW

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
@twangboy Looks like some of your new tests need to be looked at.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

File: salt/modules/mac_xattr.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def list(path, hex=False):
     cmd = 'xattr "{0}"'.format(path)
     ret = __salt__['cmd.run'](cmd)
 
-    if "No such file" in ret:
+    if "No such file" in ret or not ret:
         return None
```
Comment (cachedout) [inline salt/modules/mac_xattr.py]: Please don't return None here. I think @rallytime has already discussed this with you in another context, so I'm guessing you'll come around to this one.

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
@twangboy The unit tests for this file are now failing for `unit.modules.mac_xattr_test`.

**Comment by twangboy** [comment]
Rebased

File: salt/modules/mac_xattr.py [hunk 2]
```diff
@@ -28,34 +32,36 @@ def list(path, hex=False):
...
     hex_flag = ""
     if hex:
         hex_flag = "-x"
     cmd = 'xattr "{0}"'.format(path)
-    ret = __salt__['cmd.run'](cmd)
+    try:
+        ret = salt.utils.mac_utils.execute_return_result(cmd)
+    except CommandExecutionError as exc:
+        if 'No such file' in exc.strerror:
+            raise CommandExecutionError('File not found: {0}'.format(path))
```
Comment (cachedout) [inline salt/modules/mac_xattr.py]: Same thing here. We need to document this.

File: salt/modules/mac_xattr.py [hunk 3]
```diff
@@ -28,34 +32,36 @@ def list(path, hex=False):
     '''
     List all of the extended attributes on the given file/directory
 
+    :param str path: The file(s) to get attributes from
+
+    :param bool hex: return the values with forced hexadecimal values
```
Comment (cachedout) [inline salt/modules/mac_xattr.py]: I'm not sure why this value would be named?

File: salt/modules/mac_xattr.py [hunk 4]
```diff
@@ -28,34 +32,36 @@ def list(path, hex=False):
...
+        ret = salt.utils.mac_utils.execute_return_result(cmd)
+    except CommandExecutionError as exc:
+        if 'No such file' in exc.strerror:
+            raise CommandExecutionError('File not found: {0}'.format(path))
+        return {'Unknown Error': format(exc.strerror)}
 
-    if "No such file" in ret:
-        return None
+    if not ret:
+        return {}
```
Comment (cachedout) [inline salt/modules/mac_xattr.py]: This return is not a bool as described in this function's docstring.

File: salt/modules/mac_xattr.py [hunk 5]
```diff
@@ -64,92 +70,116 @@ def read(path, attribute, hex=False):
...
-    return __salt__['cmd.run'](cmd)
+
+    try:
+        ret = salt.utils.mac_utils.execute_return_result(cmd)
+    except CommandExecutionError as exc:
+        if 'No such file' in exc.strerror:
+            raise CommandExecutionError('File not found: {0}'.format(path))
+        if 'No such xattr' in exc.strerror:
+            raise CommandExecutionError('Attribute not found: {0}'.format(attribute))
+        return 'Unknown Error: {0}'.format(exc.strerror)
```
Comment (cachedout) [inline salt/modules/mac_xattr.py]: This also directly contradicts the type of return you have documented that this function will return.

File: salt/modules/mac_xattr.py (hunk 4 above)
Comment (twangboy) [inline salt/modules/mac_xattr.py]: Added documentation

File: salt/modules/mac_xattr.py (hunk 2 above)
Comment (twangboy) [inline salt/modules/mac_xattr.py]: Added documentation... it's supposed to return a dictionary.

**Pull request merged**